pkg/ddevapp/composer.go

```diff
@@ -20,8 +20,9 @@ func (app *DdevApp) Composer(args []string) (string, string, error) {
 	// Prevent Composer from debugging when Xdebug is enabled
 	env := []string{"XDEBUG_MODE=off"}
 	// Let Composer know which binary to run from the PATH
+	// And expand `~` in the PATH, because it is not expanded in this situation
 	path, _, err := app.Exec(&ExecOpts{
-		Cmd: "echo $PATH",
+		Cmd: `echo $PATH | sed "s|~|$HOME|g"`,
```

Comment:
I wonder if homedir.Expand() could be used successfully here. Probably not, since this is in the context of the container. But transformations like this are notoriously fragile. 

(I don't understand what homedir.Expand() replaces the ~ with. It it's $HOME, it's good, I don't think it is.)

Reply:
This PR is not only about Composer binary override, but also if in the future we have some commands that will work just like `ddev composer` with `PATH`.

This entry in `PATH=~/bin` should really be `PATH=$HOME/bin`, then we don't need these ugly workarounds.

I'll also look into the `PATH` problem with subshells while I'm here:

- https://github.com/ddev/ddev/issues/6628#issuecomment-2428795844